simple_bloom_reachability_map.go

```diff
@@ -0,0 +1,43 @@
+package node
+
+import (
+	"bloom"
+	"fmt"
+)
+
+type SimpleBloomReachabilityMap struct {
+	f *bloom.BloomFilter
+}
+
+func NewSimpleBloomReachabilityMap() ReachabilityMap {
+	m := SimpleBloomReachabilityMap{f: bloom.New(1000, 4)}
+	return m
+}
+
+func (m SimpleBloomReachabilityMap) IsReachable(n NodeAddress) bool {
+	entry := []byte(n)
+	res := m.f.Test(entry)
+	return res
+}
+
+func (m SimpleBloomReachabilityMap) AddEntry(n NodeAddress) {
+	entry := []byte(n)
+	m.f.Add(entry)
```

Comment:
No huge preference for inline vs not. I was thinking we would just evolve
this one.

On Fri, Mar 20, 2015, 01:27 Dimitar Dimitrov <notifications@github.com>
wrote:

> In simple_bloom_reachability_map.go
> <https://github.com/AutoRoute/node/pull/5#discussion_r26819479>:
>
> > +}
> > +
> > +func NewSimpleBloomReachabilityMap() ReachabilityMap {
> > +	m := SimpleBloomReachabilityMap{f: bloom.New(1000, 4)}
> > +	return m
> > +}
> > +
> > +func (m SimpleBloomReachabilityMap) IsReachable(n NodeAddress) bool {
> > +	entry := []byte(n)
> > +	res := m.f.Test(entry)
> > +	return res
> > +}
> > +
> > +func (m SimpleBloomReachabilityMap) AddEntry(n NodeAddress) {
> > +	entry := []byte(n)
> > +	m.f.Add(entry)
>
> Is there a reason to do it inline? personally I think it's easier to read
> through when it's like this, is inline more widely used ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AutoRoute/node/pull/5/files#r26819479>.
>